### DISCUSSION TITLE
Building the Project

### DISCUSSION THREAD

**Discussion by Burockhan93** [seigtm/meov#20] [General]
Hey guys! I am really interested in your meov project and wanted to look at it for myself. Unfortunately i couldnt bring CMake to work. I donw know if you are still supporting this project and could help me but i would really appreciate it. 

Here is the entire build log.

-- Building for: Visual Studio 17 2022
-- The CXX compiler identification is MSVC 19.34.31933.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at dependencies/CMakeLists.txt:15 (find_package):
  By not providing "Findassimp.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "assimp", but
  CMake did not find one.

  Could not find a package configuration file provided by "assimp" with any
  of the following names:

    assimpConfig.cmake
    assimp-config.cmake

  Add the installation prefix of "assimp" to CMAKE_PREFIX_PATH or set
  "assimp_DIR" to a directory containing one of the above files.  If "assimp"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  dependencies/CMakeLists.txt:28 (add_dependency)

CMake Warning at dependencies/CMakeLists.txt:15 (find_package):
  By not providing "Findglbinding.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "glbinding", but CMake did not find one.

  Could not find a package configuration file provided by "glbinding" with
  any of the following names:

    glbindingConfig.cmake
    glbinding-config.cmake

  Add the installation prefix of "glbinding" to CMAKE_PREFIX_PATH or set
  "glbinding_DIR" to a directory containing one of the above files.  If
  "glbinding" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  dependencies/CMakeLists.txt:28 (add_dependency)

CMake Warning at dependencies/CMakeLists.txt:15 (find_package):
  By not providing "Findglm.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "glm", but
  CMake did not find one.

  Could not find a package configuration file provided by "glm" with any of
  the following names:

    glmConfig.cmake
    glm-config.cmake

  Add the installation prefix of "glm" to CMAKE_PREFIX_PATH or set "glm_DIR"
  to a directory containing one of the above files.  If "glm" provides a
  separate development package or SDK, be sure it has been installed.
Call Stack (most recent call first):
  dependencies/CMakeLists.txt:28 (add_dependency)

CMake Warning at dependencies/CMakeLists.txt:15 (find_package):
  By not providing "Findplog.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "plog", but
  CMake did not find one.

  Could not find a package configuration file provided by "plog" with any of
  the following names:

    plogConfig.cmake
    plog-config.cmake

  Add the installation prefix of "plog" to CMAKE_PREFIX_PATH or set
  "plog_DIR" to a directory containing one of the above files.  If "plog"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  dependencies/CMakeLists.txt:28 (add_dependency)

-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - not found
-- Found Threads: TRUE
CMake Warning (dev) at C:/Program Files/CMake/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (SDL) does
  not match the name of the calling package (sdl).  This can lead to problems
  in calling code that expects `find_package` result variables (e.g.,
  `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.29/Modules/FindSDL.cmake:224 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  dependencies/CMakeLists.txt:15 (find_package)
  dependencies/CMakeLists.txt:28 (add_dependency)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Could NOT find SDL (missing: SDL_LIBRARY SDL_INCLUDE_DIR)
-- Load module: benchmark
-- Configuring done (2.9s)
CMake Error at cmake/FindPackages/Findimgui.cmake:19 (add_library):
  Cannot find source file:

    F:/Programming/Projects/meov/dependencies/imgui/imgui_demo.cpp
Call Stack (most recent call first):
  dependencies/CMakeLists.txt:15 (find_package)
  dependencies/CMakeLists.txt:28 (add_dependency)

CMake Error at cmake/FindPackages/Findimgui_filedialog.cmake:7 (add_library):
  Cannot find source file:

    F:/Programming/Projects/meov/dependencies/imgui_filedialog/ImGuiFileDialog.cpp
Call Stack (most recent call first):
  dependencies/CMakeLists.txt:15 (find_package)
  dependencies/CMakeLists.txt:28 (add_dependency)

CMake Error at cmake/FindPackages/Findimgui.cmake:19 (add_library):
  No SOURCES given to target: deps_imgui
Call Stack (most recent call first):
  dependencies/CMakeLists.txt:15 (find_package)
  dependencies/CMakeLists.txt:28 (add_dependency)

CMake Error at cmake/FindPackages/Findimgui_filedialog.cmake:7 (add_library):
  No SOURCES given to target: deps_imgui_filedialog
Call Stack (most recent call first):
  dependencies/CMakeLists.txt:15 (find_package)
  dependencies/CMakeLists.txt:28 (add_dependency)

CMake Generate step failed.  Build files cannot be regenerated correctly.

**Comment by seigtm** [comment]
Hello!

Please, try this (_in the project's top-level directory_):

```bash
git submodule update --init
```

Then try to configure and build project with CMake

**Comment by Burockhan93** [comment]
Hey thanks for the quick response! Updating dependencies have definitely brought the building process a step forward, but i am not quite there yet .) I am not sure what type of configuration i needed to make. I just built it from the main directory where the CFileList.txt was. Here is current version of the log

CMake Warning:
  Ignoring extra path from command line:

   "F:/Programming/Projects/meov/build"


-- Building for: Visual Studio 17 2022
-- The CXX compiler identification is MSVC 19.34.31933.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at dependencies/CMakeLists.txt:15 (find_package):
  By not providing "FindCMakeFiles.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "CMakeFiles", but CMake did not find one.

  Could not find a package configuration file provided by "CMakeFiles" with
  any of the following names:

    CMakeFilesConfig.cmake
    cmakefiles-config.cmake

  Add the installation prefix of "CMakeFiles" to CMAKE_PREFIX_PATH or set
  "CMakeFiles_DIR" to a directory containing one of the above files.  If
  "CMakeFiles" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  dependencies/CMakeLists.txt:28 (add_dependency)


-- The C compiler identification is MSVC 19.34.31933.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Shared libraries disabled
-- compiling zlib from sources
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of off64_t
-- Check size of off64_t - failed
-- Looking for fseeko
-- Looking for fseeko - not found
-- Looking for unistd.h
-- Looking for unistd.h - not found
-- Enabled importer formats: AMF 3DS AC ASE ASSBIN B3D BVH COLLADA DXF CSM HMP IRRMESH IQM IRR LWO LWS M3D MD2 MD3 MD5 MDC MDL NFF NDO OFF OBJ OGRE OPENGEX PLY MS3D COB BLEND IFC XGL FBX Q3D Q3BSP RAW SIB SMD STL TERRAGEN 3D X X3D GLTF 3MF MMD
-- Disabled importer formats:
-- Enabled exporter formats: OBJ OPENGEX PLY 3DS ASSBIN ASSXML M3D COLLADA FBX STL X X3D GLTF 3MF PBRT ASSJSON STEP
-- Disabled exporter formats:
-- Treating all warnings as errors (for assimp library only)
CMake Deprecation Warning at dependencies/glbinding/CMakeLists.txt:7 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Warning (dev) at dependencies/glbinding/CMakeLists.txt:47 (include):
  Policy CMP0120 is not set: The WriteCompilerDetectionHeader module is
  removed.  Run "cmake --help-policy CMP0120" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at C:/Program Files/CMake/share/cmake-3.29/Modules/WriteCompilerDetectionHeader.cmake:380 (message):
  The WriteCompilerDetectionHeader module will be removed by policy CMP0120.
  Projects should be ported away from the module, perhaps by bundling a copy
  of the generated header or using a third-party alternative.
Call Stack (most recent call first):
  dependencies/glbinding/CMakeLists.txt:47 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- External KHRplatform
-- Lib glbinding
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Failed
-- Performing Test COMPILER_HAS_DEPRECATED
-- Performing Test COMPILER_HAS_DEPRECATED - Success
-- Lib glbinding-aux
Tool glcontexts skipped: glfw3 not found
Tool glfunctions skipped: glfw3 not found
Tool glinfo skipped: glfw3 not found
Tool glisdeprecated skipped: glfw3 not found
-- Tool glmeta
Tool glqueries skipped: glfw3 not found
CMake Deprecation Warning at dependencies/glm/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at dependencies/glm/CMakeLists.txt:2 (cmake_policy):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at dependencies/plog/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Error at dependencies/sdl/CMakeLists.txt:2 (message):
  Prevented in-tree build.  Please create a build directory outside of the
  SDL source code and run "cmake -S F:/Programming/Projects/meov -B ." from
  there


-- Configuring incomplete, errors occurred!

I guess it can not find the glfw3 library. Should i download it and copy somewhere?

**Reply by seigtm** [reply]
Very strange error message here:

```
CMake Error at dependencies/sdl/CMakeLists.txt:2 (message):
Prevented in-tree build. Please create a build directory outside of the
SDL source code and run "cmake -S F:/Programming/Projects/meov -B ." from
there
```

It looks like the build somehow started as in-tree.

Try to build it like that:

```bash
cd <PROJECT_ROOT_PATH>
mkdir -p build
cmake -S . -B build
cmake --build build
```

**Reply by seigtm** [reply]
And that should not be related to the `glfw3` lib.

As you can see, there were no errors, just warnings. And it's completely normal

```
-- Performing Test COMPILER_HAS_DEPRECATED - Success
-- Lib glbinding-aux
Tool glcontexts skipped: glfw3 not found
Tool glfunctions skipped: glfw3 not found
Tool glinfo skipped: glfw3 not found
Tool glisdeprecated skipped: glfw3 not found
-- Tool glmeta
Tool glqueries skipped: glfw3 not found
CMake Deprecation Warning at dependencies/glm/CMakeLists.txt:1 (cmake_minimum_required):
Compatibility with CMake < 3.5 will be removed from a future version of
CMake.
```

**Reply by seigtm** [reply]
Is everything OK now?

Did you manage to build it?

**Comment by Burockhan93** [comment]
Heyy that helped, thanks a lot. I am not that familiar with cmake used it couple times before so i couldn't understand what the matter was. Now its been built and project opens. however i can not execute it .D It says access is denied for some reason. Whether it is Debug mode or Release mode i get the same error. Done a google search but i couldnt find a solution yet. Maybe windows somehow detects thread and refuses to run it. But apart from it everything is ok. I will try to find a workaround- if you have any idea on this as well it is greatly appreciated. Thanks a lot buddy

**Reply by seigtm** [reply]
Glad to hear that!

If you'll fail to find a solution to this problem, then feel free to create another discussion, I'll definitely try to help you.

Have a good day! 😁
